delphi/app/graph.py

```diff
@@ -1,75 +1,280 @@
-"""
-Contains classes and utilities for graph operations.
-"""
 
-class GraphNode():
+def find_path(start_node, flow, people):
     """
-    Contains information about each node within the graph representation
+    find_path is a utility function that returns all that nodes that are able to carry flow from a specific source
+    node
+    :param start_node: The source node to calculate paths for
+    :param flow: List of all nodes in the network with positive flow
+    :param people: The amount of people that must be carried from the source node
+    :return: A list of all possible nodes that contribute to viable paths for start_node
     """
+    path_list = [start_node]
+    for node in path_list:
+        current_node = node
+        for edge_check in flow:
+            if edge_check.start == current_node and edge_check.end != 'ta':
+                if edge_check.flow >= people / 2:
+                    if edge_check.end not in path_list:
+                        path_list.append(edge_check.end)
+    return path_list
 
-    def __init__(self, id_, edges=[], enabled=True):
+
+class Vertex:
+    """
+    This is a class representing a node in a graph.
+
+    Attributes:
+        name(string): the label given to the node
+        source(boolean): true if source, false otherwise
+        sink(boolean): true if sink, false otherwise
+    """
+
+    def __init__(self, name, source=False, sink=False):
+        """
+         The constructor for the vertex class
+
+         :param name: the label given to the node
+         :param source:true if source, false otherwise
+         :param sink: true if sink, false otherwise
         """
-        Constructs the GraphNode object. Initialized count to 0.
+        self.name = name
+        self.source = source
+        self.sink = sink
+
 
-        Params:
-            id_: string representing id of the node (room #)
-            edges: list of GraphEdge objects with this node as the source
-            enabled: if the node is enabled or not
+class Edge:
+    """
+        This is a class representing a directed edge in a graph.
+
+        Attributes:
+            start(string): the node the edge is starting from
+            end(string): the node the edge is going to
+            capacity(int): How much flow the edge can handle
+            flow(int): How much flow the edge ends up carrying
+            returnEdge(Edge): sister edge in the residual graph
```

Comment:
/nit returnEdge is a bit of a misleading name. Renaming it to something like "backwards edge" or something similar might be a bit more straightforward